### PR TITLE
Fix Accumbounds output for functions at negative infinity

### LINKED ISSUE #23751: Fix AccumBounds based output for `limit((sin(x)**2)**(1/x), x, -oo)`
A particular set of limits 
1) Where exponent of the function has the variable `x` and would enter the following block in `pow._eval_as_leading_term()`
```
        elif e.has(x):
            lt = exp(e * log(b))
            return lt.as_leading_term(x, logx=logx, cdir=cdir)
```
2) Then would go on in `exp_eval_as_leading_term()` and have its `arg0` being computed an `AccumBounds` term
```
    def _eval_as_leading_term(self, x, logx=None, cdir=0):
        from sympy.calculus.util import AccumBounds
        arg = self.args[0].cancel().as_leading_term(x, logx=logx)
        arg0 = arg.subs(x, 0)
```
were fixed at `S.Infinity` in #23127 

The same functions don't work at `S.NegativeInfinity` because the direction of approach (cdir) for `S.NegativeInfinity` ( -1 which is different from 1 for S.Infinity) is being neglected here .
For eg 
```
>>> limit((sin(x)**2)**(1/x), x, oo)
AccumBounds(0, 1)
>>> limit((sin(x)**2)**(1/x), x, -oo)
AccumBounds(0, 1)
```
But the expected answer is `AccumBounds(1, oo)`
image

This is from https://github.com/sympy/sympy/blob/b1d380083e71c0d3dcaeaeb1d2b83fdb4359bd83/sympy/functions/elementary/exponential.py#L545
The answer needs to be inverted when we approach from a negative direction (i.e. `cdir = -1`)

Hence a diff like this needs to be applied 
```
-            return exp(arg0)
+           if re(cdir) < S.Zero:
+               return 1/exp(arg0)
+           return exp(arg0)
```
Once the result is inverted as follows, we have the correct answer
```
>>> 1/AccumBounds(0, 1)
AccumBounds(1, oo)
```
Few tests which should be checked/added for `S.NegativeInfinity` case are 
```
>>> limit((cos(x) + 1)**(1/x), x, -oo)
>>> limit((cos(x))**(1/x), x, -oo)
>>> limit((tan(x)**2)**(2/x) , x, -oo)
>>> limit((sin(x)**2)**(1/x), x, -oo)
```

### PR DESCRIPTION
**Pull request by faze-geek**
#### References to other Issues or PRs

Fixes #23751

#### Brief description of what is fixed or changed
At `S.NegativeInfinity`, limits do not give expected accumbounds output. That has been fixed here basically by returning inverse of the ouput at `S.Infinity`.

Examples on this branch - 
```
>>> limit((cos(x) + 1)**(1/x), x, -oo)
AccumBounds(1, oo)
>>> limit((sin(x)**2)**(1/x), x, -oo)
AccumBounds(1, oo)
```

#### Other comments
Relevant tests have been added, along with the tests for #18473 because of the similarity.

#### Release Notes

* series
  * Fixes leading term for exponential functions involving AccumBounds based arguments for limits being calculated at NegativeInfinity.

### CODE REVIEW

File: sympy/functions/elementary/exponential.py [hunk 1]
```diff
@@ -540,6 +540,8 @@ def _eval_as_leading_term(self, x, logx=None, cdir=0):
             # AccumBounds(-oo, 0) or AccumBounds(-oo, oo).
             # Check out function: test_issue_18473() in test_exponential.py and
             # test_limits.py for more information.
+            if re(cdir) < S.Zero:
+                return 1/exp(arg0)
```
Comment (anutosh491) [inline sympy/functions/elementary/exponential.py:544]: Looks good , maybe we could make this `exp(-arg0)` and add few test cases like
```
>>> limit((tan(x)**2)**(2/x) , x, -oo)
AccumBounds(0, oo)
```
should be enough here , will merge this then !

Comment (faze-geek) [reply sympy/functions/elementary/exponential.py:544]: Okay sure. The `ci/circleci: Build Error` here seems irrelevant to the changes.

Comment (faze-geek) [reply sympy/functions/elementary/exponential.py]: I have made the changes, could go in after the tests run.

**Comment by anutosh491** [comment]
Looks good enabling auto-merge. I have when made some changes in the release notes to make it more accurate. Thanks

**Pull request merged**